### PR TITLE
Remove legacy license screening step

### PR DESCRIPTION
**Pull request by jasonaowen**
The last of the legacy automated screening steps is the "Verify License or Certification" step, which checks each of the submitted licenses against one of the legacy external data sources. Remove it, and all of its supporting data and code. Update the license section of the admin screening UI with something more like the facility licenses, which does not have a link to automated screening results or a checkbox.

I haven't seen any documentation around how this worked, so this feels like an appropriate moment to eulogize about its design.

There were a set of rules that mapped each license type to an external data source, including the URL, the appropriate XSL transforms, and the data matching approach. The verification step ran after those rules populated the model with the data the verification step needed, and made a SOAP call to the specified external data source with the appropriately-transformed data, transformed the response, and then set a flag on the license to indicate that it had been verified.

Given how many different license types the PSM knows about, there were a lot of different external services, a rule for each, and a request XSLT and a response XSLT for each. There were also several different matching strategies used by these different licensed types, but not used outside of this surprisingly complicated feature.

Remove:
- Verify License or Certification step from the business process
  - also remove the converging gateway that disliked having only a single path into it, and instead directly connect the LEIE step to the step after the "Verification Completed" gateway
- associated jBPM work item handler
- client the work item handler used
  - since this was the last of its kind, also remove its base class. No more SOAP!
- rules that mapped license types to external data sources
  - since that was the entirety of the file, also remove the knowledge base initialization code
- matchers specified by the rules for parsing the response
- XSL for transforming requests and responses for each license type
- data type for conveying the mapping between license types and data sources
- checkbox(es) from admin UI screen
  - as well as the backing MVC data type attribute
- JAXB data type for license verification results
- "verified" attribute from JAXB representation of a license

The enrollment process now looks like this:
enrollmentprocess-image

And here is the updated admin review page, now sans license checkboxes:
remove-automated-license-screening

I was able to test this by verifying that I could still approve previously-submitted, pending enrollments, as well as submit and approve a new enrollment.

Issue #165 Remove/edit code pointing to old external-sources app
Issue #596 Reduce use of JAXB class generation
Issue #761 Save screening results in the database

### CODE REVIEW

**Comment by jasonaowen** [comment]
> The only thing that came up was that there are still explicit license checks in the cms.screening.drl file that didn't depend on an external source. Do we still need checkboxes for manual override of those?

I think what you're talking about are rules like ["Department of Health Registration As Audiologist Is Required"](https://github.com/SolutionGuidance/psm/blob/04c15e5e254106b4329c019d3724804573ac5a05/psm-app/cms-business-process/src/main/resources/cms.screening.drl#L95-L115) - is that right, @frankduncan? If so, that's making sure the providers include the required licenses; [it should happen during the enrollment process rather than at admin review time](https://github.com/SolutionGuidance/psm/issues/572), but that's a separate issue. Those checks currently cannot be overridden - maybe they should be! But that's not functionality that we've ever had.

**Comment by jasonaowen** [comment]
Updated to remove a now-unused parameter:

```diff
--- a/psm-app/cms-web/src/main/java/gov/medicaid/controllers/admin/EnrollmentController.java
+++ b/psm-app/cms-web/src/main/java/gov/medicaid/controllers/admin/EnrollmentController.java
@@ -391,8 +391,7 @@ public class EnrollmentController extends BaseController {
     @RequestMapping("/agent/enrollment/autoScreeningResult")
     public ModelAndView viewScreeningLog(
             @RequestParam("id") long id,
-            @RequestParam("type") String type,
-            @RequestParam(value = "licenseId", required = false) String licenseId
+            @RequestParam("type") String type
     ) throws PortalServiceException {
         CMSUser user = ControllerHelper.getCurrentUser();
         Enrollment enrollment = enrollmentService.getTicketDetails(user, id);
```

**Comment by frankduncan** [comment]
@jasonaowen Ah!  I think I was just confused on the purpose of the checkboxes.

**Comment by cecilia-donnelly** [comment]
I'm surprised that the NPI and SSN verification errors still show up here:

screenshot-2018-5-3 review enrollment

(Midway through review, just noting as I go.)

**Comment by jasonaowen** [comment]
> I'm surprised that the NPI and SSN verification errors still show up

@cecilia-donnelly is that on a previously-submitted enrollment? That'd be a result of running the old screening rules when it was submitted; clicking "approve" without the checkbox checked should rerun the screening rules, produce new errors, and leave the enrollment in a pending state.

**Comment by jasonaowen** [comment]
> Out of curiosity, what is the KnowledgeBase / knowledge session stuff? Does that exist elsewhere?

That's the interface with Drools; the external sources rules were (as the file name implies) exclusively dealing with mapping each license type to an external source, including its URL, request transformer, response transformer, and matching strategy. We set up a separate knowledge session for each set of rules: validation, run as providers create their enrollment; screening, run during the admin review of a pending enrollment; and, formerly, external sources, run to prepare for talking to the external sources. The first two rule sets are still in place and still used, so yes, that still exists elsewhere.

File: psm-app/cms-business-model/src/main/java/gov/medicaid/domain/rules/inference/ProviderNameMatcher.java [hunk 1]
```diff
@@ -1,168 +0,0 @@
...
-import gov.medicaid.domain.model.PropertyListType;
-import gov.medicaid.domain.model.ProviderInformationType;
-import gov.medicaid.domain.model.SearchResultItemType;
-import gov.medicaid.domain.model.SearchResultType;
-
-import java.util.Iterator;
-import java.util.List;
-
-/**
- * This matcher compares the provider name. The comparison ignores non alphanumeric characters.
```
[the remark this answers is not in the document]
Comment (jasonaowen) [reply psm-app/cms-business-model/src/main/java/gov/medicaid/domain/rules/inference/ProviderNameMatcher.java]: Hm, interesting thought!

Right now it's pretty closely tied to the JAXB enrollment model, which we're trying to move away from with the automatic screening persistence issue.

The problem you're describing is: we should search the LEIE with multiple fields, such as NPI, name, maybe address, and so on.

I think this solves a related but sufficiently different problem: different external sources provided different search interfaces and different results, and so we needed to specify how to compare the results with the enrollment application.

So, this code might be useful inspiration when we work on that problem, but I don't think it'd be easily reused.

**Pull request merged**